Pal/lib/list.h

```diff
@@ -273,7 +273,7 @@
                             (TMP) = ((CURSOR)                               \
                                      ? (CURSOR)->FIELD.next                 \
                                      : (CURSOR)),                           \
-                            true);                                          \
+                            !!(HEAD)->first);                               \
```

Comment:
IMO `true` is exactly equal here (and much simpler). If `(HEAD)->first` is false, then the loop won't even execute any iteration (look at the loop's continue condition).